metering/reports/metering-writing-custom-report-queries.adoc

```diff
@@ -0,0 +1,396 @@
+
+For the purposes of this guide, we are going to report on the `kube_deployment_status_replicas_unavailable` metric that is produced by `kube-state-metrics`, which should be available in your Prometheus instance if you followed a default {product-title} installation.
+
+To validate this, open your web browser to your Prometheus UI and in the box above the `Execute` button, enter in `kube_deployment_status_replicas_unavailable`. Then click the `Execute` button.
+A list of metrics shows up. If you have an empty list, it is possible that you do not have `kube-state-metrics` running, or there is a configuration issue with Prometheus.
+
+== Collecting additional Prometheus metrics
+
+The first thing to figure out is what question do we want to ask, and what information is needed to ask it.
+For this guide, we are going to try to answer the following question:
+
+_What is the average and total amount of time that a particular deployment's replicas are unready?_
+
+To answer this question, we are going to use the `kube_deployment_status_replicas_unavailable` metric which tells us how many unready replicas a particular deployment has at a given moment in time.
+
+Next, we need to figure out what information we care about from this metric, as a lot of it is not particularly useful to us.
+The most relevant information available in this metric is the `namespace` and `deployment` labels, and the actual value of the metric.
+
+To strip out everything besides this information, the following Prometheus query link:https://prometheus.io/docs/prometheus/latest/querying/operators/=aggregation-operators[sums] the value of the metric grouped by the `namespace` and `deployment`.
+
+[source,bash]
+----
+sum(kube_deployment_status_replicas_unavailable) by (namespace, deployment)
+----
+
+Try the above query in the Prometheus UI to get an idea of what this changes from the original metric and what the query is doing.
+
+== Writing a PrometheusMetricsImporter ReportDataSource
+
+To configure a metric to be collected, you need to create a ReportDataSouce with a `spec.prometheusMetricsImporter` section configured to use the Prometheus query of your choice.
+Save the snippet below into a file named `unready-deployment-replicas-reportdatasource.yaml`:
+
+[source,yaml]
+----
+apiVersion: metering.openshift.io/v1
+kind: ReportDataSource
+metadata:
+  name: unready-deployment-replicas
+spec:
+  prometheusMetricsImporter:
+    query: |
+      sum(kube_deployment_status_replicas_unavailable) by (namespace, deployment)
+----
+
+Creating the ReportDataSource will cause the metering operator to create a table in Presto and begin collecting metrics using the Prometheus query in the `spec.prometheusMetricsImporter.query`. Use the following command to create the ReportDataSource:
```

Comment:
I might put the _Creating the ReportDataSource _ sentence after the command as a result.